File: Lista_03/LP3/Lista_03/lampada.java
package LP3.Lista_03;

public class lampada {
    private boolean ligada; // Define a lâmpada como ligada ou desligada
    private boolean meiaLuz; // Define se a lâmpada está em meia-luz ou não
    private int luminosidade; // Define a luminosidade (brilho) da lâmpada

    public void Lampada() {
        this.ligada = false;
        this.meiaLuz = false;
        this.luminosidade = 0;
    }

    // Operação get
    public int getLuminosidade() {
        return luminosidade;
    }

    // Operações que definem a lâmpada como ligada / desligada / meia-luz
    public void acende() {
        this.ligada = true;
        this.meiaLuz = false;
        this.luminosidade = 100;
    }

    public void apaga() {
        this.ligada = false;
        this.meiaLuz = false;
        this.luminosidade = 0;
    }

    public void meiaLuz() {
        if (this.luminosidade > 0) {
            this.meiaLuz = true;
        } else {
            this.meiaLuz = false;
        }
    }

    // Operação que faz a validação do valor da luminosidade
    private boolean validaLuminosidade(int num) {
        if (num >= 0 && num <= 100)
            return true;
        else
            return false;
    }

    // Operações set
    public void setLuminosidade(int num) {
        if (validaLuminosidade(num)) {
            this.luminosidade = num;
        } else {
            System.out.println("Invalid number!!");
            this.luminosidade = 0;
        }
    }

    public void setEstado() {
        if (this.luminosidade == 0) {
            this.ligada = false;
            this.meiaLuz = false;
        } else if (this.luminosidade == 100) {
            this.ligada = true;
            this.meiaLuz = false;
        } else {
            this.ligada = true;
            this.meiaLuz = true;
        }
    }

    public void printaInformações() {
        if (this.ligada = false) {
            System.out.println("A lâmpada está desligada");
        } else if (this.ligada == true && this.meiaLuz == true) {
            System.out.println("A lâmpada está ligada e com 100% de seu brilho");
        } else {
            System.out.println("A lâmpada está em meia-luz e com " + this.luminosidade + "% de seu brilho");
        }
    }
}

File: Lista_05/Lampada.py
class Lampada:
    # Métodos especiais __new__ e __init__
    def __new__(cls):
        instance = super().__new__(cls)
        instance.ligada = False
        instance.meiaLuz = False
        instance.luminosidade = 0
        return instance

    def getLuminosidade(self):
        return self.luminosidade

    def acende(self):
        self.ligada = True
        self.meiaLuz = False
        self.luminosidade = 100

    def apaga(self):
        self.ligada = False
        self.meiaLuz = False
        self.luminosidade = 0

    def meiaLuz(self):
        if self.luminosidade > 0:
            self.meiaLuz = True
        else:
            self.meiaLuz = False

    def validaLuminosidade(self, num):
        if 0 <= num <= 100:
            return True
        else:
            return False

    def setLuminosidade(self, num):
        if self.validaLuminosidade(num):
            self.luminosidade = num
        else:
            print("Invalid number!!")
            self.luminosidade = 0

    def setEstado(self):
        if self.luminosidade == 0:
            self.ligada = False
            self.meiaLuz = False
        elif self.luminosidade == 100:
            self.ligada = True
            self.meiaLuz = False
        else:
            self.ligada = True
            self.meiaLuz = True

    def printaInformacoes(self):
        if self.ligada == False:
            print("A lâmpada está desligada")
        elif self.ligada == True and self.meiaLuz == True:
            print("A lâmpada está ligada e com 100% de seu brilho")
        else:
            print(f"A lâmpada está em meia-luz e com {self.luminosidade}% de seu brilho")